pkg/controllers/jobset_controller_test.go

```diff
@@ -1205,3 +1217,152 @@ func makeJob(args *makeJobArgs) *testutils.JobWrapper {
+			expectServiceCreate:            true,
+			expectServiceName:              "test-jobset",
+			expectPublishNotReadyAddresses: true,
+		},
+		{
+			name:                           "service does not exist and should be created, subdomain set",
+			jobSet:                         testutils.MakeJobSet(jobSetName, ns).EnableDNSHostnames(true).NetworkSubdomain("test-subdomain").Obj(),
+			expectServiceCreate:            true,
+			expectServiceName:              "test-subdomain",
+			expectPublishNotReadyAddresses: true,
+		},
+		{
+			name:                           "service does not exist and should be created, publishNotReadyAddresses is false",
+			jobSet:                         testutils.MakeJobSet(jobSetName, ns).EnableDNSHostnames(true).PublishNotReadyAddresses(false).Obj(),
+			expectServiceCreate:            true,
+			expectServiceName:              "test-jobset",
+			expectPublishNotReadyAddresses: false,
+		},
+		{
+			name:                           "service does not exist and should be created, publishNotReadyAddresses is true",
+			jobSet:                         testutils.MakeJobSet(jobSetName, ns).EnableDNSHostnames(true).PublishNotReadyAddresses(true).Obj(),
+			expectServiceCreate:            true,
+			expectServiceName:              "test-jobset",
+			expectPublishNotReadyAddresses: true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			var servicesCreated int
+			_, ctx := ktesting.NewTestContext(t)
+			scheme := runtime.NewScheme()
+			utilruntime.Must(jobset.AddToScheme(scheme))
+			utilruntime.Must(corev1.AddToScheme(scheme))
+			fakeClientBuilder := fake.NewClientBuilder().WithScheme(scheme).WithInterceptorFuncs(interceptor.Funcs{
+				Create: func(ctx context.Context, client client.WithWatch, obj client.Object, opts ...client.CreateOption) error {
+					if tc.expectErr {
+						return errors.New("unexpected error")
+					}
+					if !tc.expectServiceCreate {
+						t.Fatal("unexpected service creation")
+					}
+					svc := obj.(*corev1.Service)
+					if svc.Name != tc.expectServiceName {
+						t.Errorf("expected service name to be %q, got %q", tc.expectServiceName, svc.Name)
+					}
+					if len(svc.OwnerReferences) != 1 {
+						t.Error("expected service to have owner reference set")
+					}
+					expectedOwnerRef := metav1.OwnerReference{
+						APIVersion:         "jobset.x-k8s.io/v1alpha2",
+						Kind:               "JobSet",
+						Name:               "test-jobset",
+						Controller:         ptr.To(true),
+						BlockOwnerDeletion: ptr.To(true),
+					}
+					if diff := cmp.Diff(expectedOwnerRef, svc.OwnerReferences[0]); diff != "" {
+						t.Errorf("unexpected service owner reference value (+got/-want): %s", diff)
+					}
+					if svc.Spec.ClusterIP != corev1.ClusterIPNone {
+						t.Errorf("expected service to have ClusterIP None, got %s", svc.Spec.ClusterIP)
+					}
+					selector, ok := svc.Spec.Selector[jobset.JobSetNameKey]
```

Comment:
The variable name `selector` implies it contains the full key-value pair of a label selector. 

Let's use `selectorValue` or something similar so it's clear it only contains the node label value.